airflow/container_runtime_test.go

```diff
@@ -0,0 +1,75 @@
+package airflow
+
+// MockFileChecker is a mock implementation of FileChecker for tests.
+type MockFileChecker struct {
+	existingFiles map[string]bool
+}
+
+// Exists is just a mock for os.Stat(). In our test implementation, we just check
+// if the file exists in the list of mocked files for a given test.
+func (m MockFileChecker) Exists(path string) bool {
+	return m.existingFiles[path]
+}
+
+// TestGetContainerRuntimeBinary runs a suite of tests against GetContainerRuntimeBinary,
+// using the MockFileChecker defined above.
+func (s *Suite) TestGetContainerRuntimeBinary() {
+	tests := []struct {
+		name      string
+		pathEnv   string
+		binary    string
+		mockFiles map[string]bool
+		expected  bool
+	}{
+		{
+			name:    "Find docker",
+			pathEnv: "/usr/local/bin:/usr/bin:/bin",
```

Comment:
The tests should run on windows because the `MockFileChecker` just looks for a match in the `mockFiles` map. I originally had the `os.Stat` call directly in the function here, but that wouldn't work even across unixy machines since it was actually checking the filesystem which could differ drastically.

Reply:
While we're on the topic though, we should probably think about some automated way to run the tests on Windows (a future PR). @sam-awezec was digging into the traffic coming through our onboarding flow and we're seeing a pretty even 50/50 split between mac and windows. Seems like the winget release process could use some love too.